scs2-simulation/src/main/java/us/ihmc/scs2/simulation/screwTools/SimMultiBodySystemTools.java

```diff
@@ -1206,4 +1203,141 @@ private static int insertJointsTau(JointBasics[] joints, int startIndex, DMatrix
+   }
+
+   public static YoMatrix createMatrixForJointsAcceleration(String name, String description, SimMultiBodySystemReadOnly system, YoRegistry registry)
+   {
+      // Acceleration has the same dimensions / joint indices as torque
+      return createYoMatrixForJointsTau(name, description, system, registry);
+   }
+
+   public static YoMatrix createYoMatrixForJointsTau(String name, String description, SimMultiBodySystemReadOnly system, YoRegistry registry)
+   {
+      List<? extends  JointReadOnly> joints = system.getJointMatrixIndexProvider().getIndexedJointsInOrder();
+
+      List<String> rowNames = new ArrayList<>();
+      for (JointReadOnly joint : joints)
+      {
+         if (joint instanceof OneDoFJointReadOnly)
+         {
+            rowNames.add(joint.getName());
+
+         }
+         else if (joint instanceof SphericalJointReadOnly)
+         {
+            rowNames.add(joint.getName() + "_ax");
+            rowNames.add(joint.getName() + "_ay");
+            rowNames.add(joint.getName() + "_az");
+
+         }
+         else if (joint instanceof PlanarJointReadOnly)
+         {
+            rowNames.add(joint.getName() + "_ay");
+            rowNames.add(joint.getName() + "_lx");
+            rowNames.add(joint.getName() + "_lz");
+
+         }
+         else if (joint instanceof SixDoFJointReadOnly)
+         {
+            rowNames.add(joint.getName() + "_ax");
+            rowNames.add(joint.getName() + "_ay");
+            rowNames.add(joint.getName() + "_az");
+            rowNames.add(joint.getName() + "_lx");
+            rowNames.add(joint.getName() + "_ly");
+            rowNames.add(joint.getName() + "_lz");
+
+         }
+         else  // any other type of joint
+         {
+            if (joint.getDegreesOfFreedom() > 1)
+            {
+               for (int i = 0; i < joint.getDegreesOfFreedom(); i++)
+                  rowNames.add(joint.getName() + "_" + i);
+            }
+            else
+            {
+               rowNames.add(joint.getName());
+            }
+         }
+      }
+
+      String[] rowNamesArray = rowNames.toArray(new String[0]);
+      return new YoMatrix(name, description, rowNamesArray.length, 1, rowNamesArray, null, registry);
+   }
+
+   public static YoMatrix createMatrixForJointsState(String name, String description, SimMultiBodySystemReadOnly system, SimJointStateType state, YoRegistry registry)
+   {
+      switch(state)
+      {
+         case CONFIGURATION:
+            return createYoMatrixForJointsConfiguration(name, description, system, registry);
+         case VELOCITY, ACCELERATION, EFFORT:
+            // Velocity, acceleration, and effort all have the same dimensions / joint indices
+            return createYoMatrixForJointsTau(name, description, system, registry);
+         default:
+            throw new RuntimeException("Unexpected value for state: " + state);
+      }
+   }
```

Comment:
⬆️ methods for creating `YoMatrix` for diffferent joint states. I had trouble coming up with a name for the two groups:
1. Configuration (different dimension due to possible quaternions)
2. Velocity, Acceleration, Effort (all the same dimensions)
So I just sneakily made velocity and acceleration call the torque method internally. This may be poor.

Reply:
Done